include/beman/execution26/detail/intrusive_queue.hpp

```diff
@@ -0,0 +1,59 @@
+// include/beman/execution26/detail/intrusive_queue.hpp               -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_INTRUSIVE_QUEUE
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_INTRUSIVE_QUEUE
+
+#include <cassert>
+#include <utility>
+
+namespace beman::execution26::detail {
+
+template <auto Next>
+class intrusive_queue;
+
+//! @brief  This data structure is an intrusive queue that is not thread-safe.
+template <class Item, Item* Item::*Next>
+class intrusive_queue<Next> {
+  public:
+    ~intrusive_queue() { assert(!head_); }
+
+    //! @brief  Pushes an item to the queue.
+    auto push(Item* item) noexcept -> void {
+        if (last_) {
+            assert(head_);
+            last_->*Next = item;
+        } else {
+            assert(!head_ && !last_);
+            head_ = item;
+            last_ = item;
```

Comment:
It seems this assignment needs to be outside of the `else`-branch: if `last_` wasn't `nullptr` before the operation, `item` also becomes the new `last_` element.

Reply:
I guess this class implementation is not sound and wrong for the general case. In split I just call pop and call it a day but it can't stay like this. I might make this a class containing single pointer only that you can iterate through.

Good catch.

Reply:
I think `split` just needs a list of items and the order doesn't matter. Moving it to an `intrusive_stack` rather than an `intrusive_queue` would be fine. I guess the use in `split` works with the current implementation but other uses, if they energe, might not.